EDU-18421 Using CSS - Sales App extensibility (#2682)
vtexdocs/dev-portal-content: docs/guides/VTEX-Sales-App/css-styling-in-vtex-sales-app-extensibility.md

---
title: "CSS styling in VTEX Sales App Extensibility"
slug: "css-styling-in-vtex-sales-app-extensibility"
hidden: false
excerpt: "Learn how to style extensions using CSS in VTEX Sales App Extensibility."
createdAt: "2026-05-25T00:00:00.000Z"
---

> ⚠️ This feature is in beta, and we're working to improve it. If you have any questions, please contact our [Support](https://help.vtex.com/en/support).

> Once you have implemented a [VTEX Sales App Extensibility](link-placeholder) project, you can use CSS to style an extension. This is an optional configuration and it can be done in two ways:

Author: O [texto original](https://beta.fast.store/sales-app/using-css) cita "CSS Variables" ao final (print), mas o link dá 404. Tem algo nesse sentido para ser referenciado ou não precisa?

<img width="946" height="155" alt="Image" src="https://github.com/user-attachments/assets/38c6a103-8f18-4319-8d12-b8ff289bf587" />

Reviewer: Também não encontrei, Isa. Acredito que não precisa, não sei o que ocorreu com essa página.

> Once you have implemented a [VTEX Sales App Extensibility](link-placeholder) project, you can use CSS to style an extension. This is an optional configuration and it can be done in two ways:

Reviewer 2: ```suggestion
Once you have [created a Sales App extension](https://developers.vtex.com/docs/guides/creating-sales-app-extensions), you can style your component using CSS. This configuration is optional, and you can do it in two ways:
```

> - Using [`cssmodules`](https://github.com/css-modules/css-modules) to access classes through the object automatically generated by the bundler. This is the recommended approach.

Reviewer 2: ```suggestion
- Using [CSS Modules](https://github.com/css-modules/css-modules) to import CSS classes as an object generated by the bundler, so you can apply locally scoped styles in your component. This is the recommended approach.
```

> - Importing plain `.css` files so class names apply globally (e.g. `className='btn'`).

Reviewer 2: ```suggestion
- Importing plain `.css` files to apply classes globally (for example, `className='btn'`).
```

## Scope and limitations

Using CSS is recommended exclusively for styling your extensions. CSS files have access to the global scope where they are loaded, meaning they can technically target and style other elements of the Sales App core.

> However, using global CSS to style elements outside of your extension, relying on selectors like `tag`, `class`, `id`, or any other combination not specific to the extension, is discouraged for the following reasons:

Reviewer 2: ```suggestion
Using CSS is recommended exclusively for styling your extensions. CSS files have access to the global scope where they're loaded, meaning they can technically target and style other elements of the Sales App core.

> ⚠️ Any styling applied to core elements is subject to breaking changes due to bug fixes, new features, and other updates.

Avoid using global CSS to target elements outside your extension, such as with `tag`, `class`, or `id` selectors, for the following reasons:
```

- Future releases may include sandboxing mechanisms that isolate CSS and/or extensions from the core application's global context (e.g., DOM).
- Using selectors like classes or IDs on core Sales App elements (e.g., the "Continue" button, the cart list, header, logo) is not supported, as Sales App generates unique IDs for classes applied to its elements.

> - Relying on the markup to create selectors with the goal of customizing the core Sales App is risky, as the markup may change with core updates (e.g., adding or removing elements in the DOM).

Reviewer 2: ```suggestion
- Future releases may include sandboxing mechanisms that isolate CSS and/or extensions from the core application's global context (for example, DOM).
- Using selectors like classes or IDs on core Sales App elements (for example, the "Continue" button, the cart list, header, logo) is not supported, as Sales App generates unique IDs for classes applied to its elements.
- Relying on the markup to create selectors with the goal of customizing the core Sales App is risky, as the markup may change with core updates (for example, adding or removing elements in the DOM).
```

With that in mind, CSS should only be used for styling the extension itself. Any styling applied to core elements is subject to breaking changes due to bug fixes, new features, and other updates.

> 

Reviewer 2: ```suggestion
```

> > ℹ️ Visual changes, such as the primary brand color, are currently done through configuration. Other visual customizations (like fonts, borders, etc.) will be possible in future releases.

Reviewer 2: ```suggestion
> ℹ️ Visual changes, such as the primary brand color, are currently done through configuration.
```

Reviewer 2: I suggest removing the mention of future releases, as it may create expectations for something we don’t know when (or even if) it'll become available.

> ## Styling extensions using CSS modules (recommended)

Reviewer 3: Na verdade hoje o projeto de extensão para o Sales App não suporta CSS Modules. Apenas plain CSS. É uma melhoria que devemos fazer logo, pois é de fato o padrão atual. Mas hoje não suporta

With CSS Modules, classes are imported as an object, and you can apply them to your components dynamically, ensuring a clean separation of styles.

This is the recommended approach because it ensures that class names are scoped locally to the component, avoiding potential conflicts with other styles in the global namespace.

CustomMessage.tsx example:

```tsx
  // CustomMessage.tsx
  import styles from './custom-message.module.css';

  export const CustomMessage = () => {
    return (
      <p className={styles.footer}>
        Extension example
      </p>
    );
  };
```

custom-message.module.css example:

```css
  /* custom-message.module.css */
  .custom-message {
    background-color: black;
    color: white;
  }
```

> ℹ️ When using CSS Modules, the bundler automatically creates unique class names, so there's no risk of name conflicts between classes.

## Styling extensions with plain CSS imports

> In this approach, you will include `.css` files in the relevant TypeScript file. You can either create a single file and import it once in `src/index.tsx`, or create component-specific files and import them in the component's file.

Reviewer 2: ```suggestion
In this approach, you'll include `.css` files in the relevant `.tsx` file. You can either create a single file and import it once in `src/index.tsx`, or create component-specific files and import them in the component's file.
```

Single CSS file example:

> ```typescript

Reviewer 2: ```suggestion
```tsx
```

  // src/index.tsx
  import { defineExtensions } from '@vtex/sales-app'
  import './styles.css'
```

Multiple CSS files examples:

> ```typescript

Reviewer 2: ```suggestion
```tsx
```

  // CustomMessage.tsx
  import './custom-message.css'
  const CustomMessage = () => {...}
```

> ```typescript

Reviewer 2: ```suggestion
```tsx
```
